File: Python/binary_search.py
itemList = [1,2,3,4,5,6,7,8,9,10]
item = 2

def binary_search (itemList, item):
    low = 0
    high = len(itemList) - 1
    
    while (low <= high):
        mid = (low + high) // 2
        guess = itemList[mid]
        if (guess == item):
            return mid
        elif (guess > item):
            high = mid - 1
        else:
            low = mid + 1
    return None

info = binary_search(itemList, item)
print(info)

File: JavaScript/binary_search.js
let item_list = [1,2,3,4,5,6,7,8,9]
item = 8

function binary_search (list, item) {
    low = 0;
    high = item_list.length - 1;
    while (low <= high) {
        mid = Math.floor((high + low) / 2);
        guess = item_list[mid];
        if (guess == item) {
            return mid;
        } else if (item < guess){
            high = mid - 1;
        } else {
            low = mid + 1
        }
    }
    return null;
}

info = binary_search(item_list, item);
console.log(info);